src/components/ActionSheetAwareScrollView/ActionSheetKeyboardSpace.js

```diff
@@ -0,0 +1,237 @@
+import React, {
+    useContext, useEffect, useMemo,
+} from 'react';
+import {Dimensions, Keyboard} from 'react-native';
+import Reanimated, {
+    useWorkletCallback,
+    Easing,
+    KeyboardState,
+    useAnimatedKeyboard,
+    useAnimatedStyle, useDerivedValue, withTiming,
+} from 'react-native-reanimated';
+import {useSafeAreaInsets} from 'react-native-safe-area-context';
+import styles from '../../styles/styles';
+import {Actions, States, ActionSheetAwareScrollViewContext} from './ActionSheetAwareScrollViewContext';
+
+const config = {
+    duration: 350,
+    easing: Easing.bezier(0.33, 0.01, 0, 1),
+};
+
+function ActionSheetKeyboardSpace(props) {
+    const safeArea = useSafeAreaInsets();
+    const keyboard = useAnimatedKeyboard();
+
+    // similar to using `global` in worklet but it's just a local object
+    const ctx = useMemo(() => ({}), []);
```

Comment:
Sorry, I'm confused why this is needed. Why can't we just do:

```js
const ctx = {};
```

?

Reply:
Is it because we want `ctx` to be preserved between renders?

Reply:
Also, I think it would be clearer to track `shouldRunAnimation` and `keyboardValue` separate v.s: putting them in the same object.

Reply:
I totally forgot about the second value, i'll put them directly into the same object instead and will use a better name than `ctx`, something like `localState`.